### DISCUSSION TITLE
Chat window missing menu items when tabbed

### DISCUSSION THREAD

**Discussion by chrisrimple** [jamulussoftware/jamulus#2896] [Ideas]
On Mac OS (Mojave), I've enabled Tabs for applications: Apple > System Preferences > Dock > Prefer Tabs = Always.  As a result, Jamulus windows appear as tabs in a single application window.  That's great, except that clicking on the Chat tab hides the File, View, and Help menus.  The missing menus should always be visible.

<img width="797" alt="image" src="https://user-images.githubusercontent.com/47258736/98054709-8f643e00-1df0-11eb-84a6-b7c69260bd8d.png">

<img width="798" alt="image" src="https://user-images.githubusercontent.com/47258736/98054721-9723e280-1df0-11eb-9462-642ec2631b8e.png">

**Comment by corrados** [comment]
As far as I understand this new Mac OS feature (since I have an old version of Mac OS, I do not have this feature available), it simply puts the windows of an application in tabs. So the menu what you see is the same as if your application was not "tabbed" and the window is selected. So, e.g., if you do not use that new Mac OS feature, start Jamulus and have the Chat window got the focus, you should also only see the Edit menu, right?

**Comment by chrisrimple** [comment]
@corrados Yes, you're correct.  But perhaps all the menus should be present even for the Chat window, for ease-of-use?

**Comment by corrados** [comment]
The menus are window specific in Jamulus, e.g.:

Jamulus main window Edit menu:
![grafik](https://user-images.githubusercontent.com/46655886/98460906-9dc3a880-21a8-11eb-96ac-b4503ff91b96.png)

Chat window Edit menu:
![grafik](https://user-images.githubusercontent.com/46655886/98460913-b3d16900-21a8-11eb-8204-b6cb3224482f.png)

**Comment by chrisrimple** [comment]
My point is that perhaps the menus should not be window-specific.  There are many multi-window applications in which all menus are always present.  :)

**Comment by ann0see** [comment]
Hi, Sorry for the maintenance noise here. I’m just into triaging issues.

I think this issue can be resolved by some code wise re-structuring, since other apps also do that. Probably something is not implemented correctly in Jamulus? 

I'm not an expert for macOS apps, but maybe some macOS dev can jump in here?

@ngocdh @emlynmac

Note: I will unsubscribe from this issue and won’t receive responses from any new comments. If you have any questions concerning maintenance, feel free to ping me.

**Comment by pljones** [comment]
The current Jamulus implementation runs with each window as an independent task.  To group them in tabs "feel wrong" from the Jamulus UI design because it _is_ wrong for the Jamulus UI design.

Switching between the Chat Window and any other window on Windows, for example, is the same as switching between two separate applications - Alt-TAB.  If you want to close the Chat Window, it's the same as closing an _application_, Alt-F4.

That design is currently intentional - "Escape" to close a subsidiary window has been deliberately prevented.  However, I do think the UI design could benefit from a rethink at some point.

**Reply by pljones** [reply]
I've moved this to a discussion as clearly more thought it involved and the behaviour isn't a bug as such.
